src/Symfony/Component/Lock/Bridge/DynamoDb/Store/DynamoDbStore.php

```diff
@@ -0,0 +1,277 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Bridge\DynamoDb\Store;
+
+use AsyncAws\DynamoDb\DynamoDbClient;
+use AsyncAws\DynamoDb\Exception\ConditionalCheckFailedException;
+use AsyncAws\DynamoDb\Input\CreateTableInput;
+use AsyncAws\DynamoDb\Input\DeleteItemInput;
+use AsyncAws\DynamoDb\Input\DescribeTableInput;
+use AsyncAws\DynamoDb\Input\GetItemInput;
+use AsyncAws\DynamoDb\Input\PutItemInput;
+use AsyncAws\DynamoDb\ValueObject\AttributeDefinition;
+use AsyncAws\DynamoDb\ValueObject\AttributeValue;
+use AsyncAws\DynamoDb\ValueObject\KeySchemaElement;
+use AsyncAws\DynamoDb\ValueObject\ProvisionedThroughput;
+use Symfony\Component\Lock\Exception\InvalidArgumentException;
+use Symfony\Component\Lock\Exception\InvalidTtlException;
+use Symfony\Component\Lock\Exception\LockAcquiringException;
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\PersistingStoreInterface;
+use Symfony\Component\Lock\Store\ExpiringStoreTrait;
+
+class DynamoDbStore implements PersistingStoreInterface
+{
+    use ExpiringStoreTrait;
+
+    private const DEFAULT_OPTIONS = [
+        'access_key' => null,
+        'secret_key' => null,
+        'session_token' => null,
+        'endpoint' => null,
+        'region' => null,
+        'table_name' => 'lock_keys',
+        'id_attr' => 'key_id',
+        'token_attr' => 'key_token',
+        'expiration_attr' => 'key_expiration',
+        'read_capacity_units' => 10,
+        'write_capacity_units' => 20,
+        'sslmode' => null,
+        'debug' => null,
+    ];
+
+    private DynamoDbClient $client;
+    private string $tableName;
+    private string $idAttr;
+    private string $tokenAttr;
+    private string $expirationAttr;
+    private int $readCapacityUnits;
+    private int $writeCapacityUnits;
+
+    public function __construct(
+        DynamoDbClient|string $clientOrUrl,
+        array $options = [],
+        private readonly int $initialTtl = 300
+    ) {
+        if ($clientOrUrl instanceof DynamoDbClient) {
+            $this->client = $clientOrUrl;
+        } else {
+            if (!class_exists(DynamoDbClient::class)) {
+                throw new InvalidArgumentException(\sprintf('You cannot use the "%s" if the DynamoDbClient is not available. Try running "composer require async-aws/dynamo-db".', __CLASS__));
+            }
+
```

Comment:
This is not necessary, since the class has its dedicated package, the `async-aws/dynamo-db` package is required by composer.

```suggestion
```

Reply:
Done